apps/root/src/pages/token-profile/historical-prices/index.tsx

```diff
@@ -0,0 +1,356 @@
+  payload?: {
+    value?: ValueType;
+    name?: NameType;
+    dataKey?: string | number;
+    payload?: DataItem;
+  }[];
+}
+
+const GraphTooltip = (props: TooltipProps) => {
+  const { payload } = props;
+  const intl = useIntl();
+
+  const firstPayload = payload && payload[0];
+
+  if (!firstPayload || !firstPayload.payload) {
+    return null;
+  }
+
+  const actions = firstPayload.payload.actions;
+
+  return (
+    <StyledTooltipContainer>
+      {actions.length === 0 ? (
+        <ContainerBox flexDirection="column" gap={1}>
+          <Typography variant="bodySmallRegular">
+            ${formatUsdAmount({ intl, amount: firstPayload.payload.price })}
+          </Typography>
+          <Typography variant="bodySmallRegular">{firstPayload.payload.name}</Typography>
+        </ContainerBox>
+      ) : (
+        actions.map(({ user, value, type, date }, key) => (
+          <ContainerBox gap={1} flexDirection="column" key={`${key}-${date}`}>
+            <ContainerBox justifyContent="space-between" gap={3}>
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                {type === TransactionEventTypes.SWAP ? (
+                  <FormattedMessage
+                    description="token-profile-historical-prices.tooltip.swapped"
+                    defaultMessage="Swapped:"
+                  />
+                ) : (
+                  <FormattedMessage
+                    description="token-profile-historical-prices.tooltip.transfered"
+                    defaultMessage="Transfered:"
+                  />
+                )}
+              </Typography>
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                <Address address={user} trimAddress />
+              </Typography>
+            </ContainerBox>
+            <ContainerBox alignItems="center" gap={1}>
+              <TokenIcon token={value.token} size={5} />
+              <Typography variant="bodySmallBold" color={({ palette: { mode } }) => colors[mode].typography.typo2}>
+                {formatCurrencyAmount({ amount: value.amount.amount, token: value.token, intl })}
+              </Typography>
+              <Typography variant="bodySmallRegular" color={({ palette: { mode } }) => colors[mode].typography.typo3}>
+                (${formatUsdAmount({ intl, amount: value.amount.amountInUSD })})
+              </Typography>
+            </ContainerBox>
+          </ContainerBox>
+        ))
+      )}
+    </StyledTooltipContainer>
+  );
+};
+
+const PeriodMap: Record<Exclude<GraphContainerPeriods, GraphContainerPeriods.all>, TimeString> = {
+  [GraphContainerPeriods.day]: '30m',
+  [GraphContainerPeriods.week]: '3h',
+  [GraphContainerPeriods.month]: '12h',
+  [GraphContainerPeriods.year]: '1w',
+};
+
+const DATA_POINTS = 200;
+
+const TokenHistoricalPrices = ({ token }: TokenHistoricalPricesProps) => {
```

Comment:
Seems like there are a looot of thinkgs on this file, could we maybe split it off into several ones so it remains shorter?

Reply:
Sure, I'll work on that now